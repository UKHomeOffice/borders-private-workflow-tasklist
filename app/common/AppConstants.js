
export default class AppConstants {
    static DASHBOARD_PATH = "/dashboard";
    static YOUR_TASKS_PATH = "/your-tasks";
    static YOUR_GROUP_TASKS_PATH = '/your-group-tasks';
    static PROCEDURES_PATH = '/procedures';
    static REPORTS_PATH = '/reports';
    static REPORT_PATH = "/report";
    static MESSAGES_PATH = "/messages";
    static CALENDAR_PATH = "/calendar";
    static START_A_PROCEDURE = "/start-a-procedure";
    static TASK_PATH="/task";
    static ADMIN_PATH="/admin";
    static SHIFT_PATH = "/shift";
    static PROCEDURE_DIAGRAM_PATH = "/procedure-diagram";
    static ONBOARD_USER_PATH="/onboard-user";
    static MANDATORY_DECLARATION_PATH="/mandatory-declarations";
    static  ONE_MINUTE = 60000;

}
