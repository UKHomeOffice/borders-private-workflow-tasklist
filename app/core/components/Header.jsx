import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {withRouter} from 'react-router-dom';
import {bindActionCreators} from 'redux';
import secureLocalStorage from '../../common/security/SecureLocalStorage';
import AppConstants from "../../common/AppConstants";

export class Header extends React.Component {

    constructor(props) {
        super(props);
        this.secureLocalStorage = secureLocalStorage;
        this.logout = this.logout.bind(this);
        this.dashboard = this.dashboard.bind(this);
    }

    dashboard(event) {
        event.preventDefault();
        this.props.history.push(AppConstants.DASHBOARD_PATH);
    }

    logout(event) {
        event.preventDefault();
        this.secureLocalStorage.removeAll();
       this.props.kc.logout();
    }

    render() {
        return <header className="govuk-header" role="banner" data-module="header">
                <div className="govuk-header__container govuk-width-container">
                    <div className="govuk-header__content" style={{width: '100%'}}>
                        <div className="govuk-grid-row">
                            <div className="govuk-grid-column-one-half">
                                <a
                                    id="dashboard"
                                    href={AppConstants.DASHBOARD_PATH}
                                    onClick={event => this.dashboard(event)}
                                    className="govuk-header__link govuk-header__link--service-name"
                                >Central Operations Platform</a>
                            </div>
                            <div className="govuk-grid-column-one-half header-nav">
                                <a
                                    id="profile"
                                    href={`${AppConstants.SUBMIT_A_FORM}/edit-your-profile`}
                                    onClick={(event) => {
                                        event.preventDefault();
                                        this.props.history.replace(`${AppConstants.SUBMIT_A_FORM}/edit-your-profile`)
                                    }}
                                    className="govuk-header__link header-nav__link"
                                >My profile</a>
                                <a
                                    id="support"
                                    href="https://support.cop.homeoffice.gov.uk/servicedesk/customer/portal/3"
                                    className="govuk-header__link header-nav__link"
                                >Support</a>
                                <a
                                    id="logout"
                                    href={"/logout"}
                                    onClick={this.logout}
                                    className="govuk-header__link header-nav__link"
                                >Sign out</a>
                            </div>
                        </div>
                    </div>
                </div>
            </header>;
    }
}

Header.propTypes = {
    history: PropTypes.shape({
        push: PropTypes.func,
    }).isRequired,
    kc: PropTypes.shape({
        logout: PropTypes.func,
    }).isRequired,
};

const mapDispatchToProps = dispatch => bindActionCreators({}, dispatch);

export default withRouter(connect(state => ({
    kc: state.keycloak,
    appConfig: state.appConfig,
}), mapDispatchToProps)(Header));
