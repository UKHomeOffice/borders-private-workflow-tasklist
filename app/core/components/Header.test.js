import React from 'react';
import { createMemoryHistory } from 'history'
import { Header } from './Header';
import AppConstants from '../../common/AppConstants';

describe('Header', () => {

  const props = {
    history: createMemoryHistory(),
    appConfig: {
      serviceDeskUrls: {
        support: 'test'
      }
    },
    kc: {
      logout: jest.fn(),
    },
    location: {
      pathname: AppConstants.DASHBOARD_PATH
    }
  }

  it('renders without crashing', () => {
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper.exists()).toBe(true);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

});
