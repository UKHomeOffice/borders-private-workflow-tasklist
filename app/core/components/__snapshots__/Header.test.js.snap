// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div>
  <header
    className="govuk-header"
    data-module="header"
    role="banner"
  >
    <div
      className="govuk-header__container govuk-width-container"
    >
      <div
        className="govuk-header__content"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <a
              className="govuk-header__link govuk-header__link--service-name"
              href=""
              onClick={[Function]}
            >
              Central Operations Platform
            </a>
          </div>
          <div
            className="govuk-grid-column-one-third header-nav"
          >
            <a
              className="govuk-header__link header-nav__link"
              href="/start-a-procedure/edit-your-profile"
              id="profile"
            >
              My profile
            </a>
            <a
              className="govuk-header__link header-nav__link"
              href="https://support.cop.homeoffice.gov.uk/servicedesk/customer/portal/3"
              id="support"
            >
              Support
            </a>
            <a
              className="govuk-header__link header-nav__link"
              href=""
              id="logout"
              onClick={[Function]}
            >
              Sign out
            </a>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
